libflux/fluxdoc/SPEC.md

```diff
@@ -0,0 +1,284 @@
+# Writing Flux Doc Comments
+
+Flux will treat any uninterrupted series of line comments immediately preceding a function definition, package clause, builtin statement, or any declaration of an option or constant, as a doc comment.
+
+A flux doc comment consists of multiple sections, which are defined in this document. Each section should be separated from other sections by a single blank line comment. Unless otherwise noted, each section is required, and the sections must be arranged in the order in which they are presented in this document.
+
+## Documenting functions
+
+### Name and Description
+
+Doc comments for functions should begin with the function name, wrapped in backticks, followed by a brief description of what the function does or how it is used. The name and description should read as one or more full sentences.
+
+_Example:_
+
+```
+// `join` merges two input streams into a single output stream based on columns
+// with equal values.
+```
+
+### Extra Details
```

Comment:
Can you expand this section with an example? I didn't understand what was meant here.

Reply:
Yes, I will do that.